lib/rubygems/config_file.rb

```diff
@@ -210,22 +210,32 @@ def initialize(args)
       @hash = @hash.merge environment_config
     end
 
+    @hash.transform_keys! do |k|
+      # These keys didn't work as symbols
+      if %w[gem install gemhome gempath].include?(k)
```

Comment:
From the comment in lines 15-16 in this file:

> gemrc is a YAML file that uses strings to match gem command arguments and
> symbols to match RubyGems options.

My understanding is that right now, confusingly, you can specify any command name (or  `gem` for all commands), and that's specified as string. For RubyGems options, they are specified as keys.

So, I think we should:

* Include all commands here, not just `install`, otherwise configurations for other commands are going to be symbolized and possible break?
* Remove or reword the comment above to no longer document this inconsistency.

Makes sense?

Reply:
I changed my proposal. I only convert `Gem::ConfigFile` accessors. I don't want to work this simple issue to `Gem.configuration` entirely.